### PR TITLE
Feature/cs-021/antrl-file-reader-by-text

### PR DESCRIPTION
**Pull request by diego-centella-jalafund**
#### Resolved task/US: ([File reader antlr by text](https://app.clickup.com/t/868amwtfg))
* * * *

## What's in this pull request?
There was implemented the antrl file reader handled by text one string that has read by one file reader

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
It was tested the file reader input, for the input string on the file reader using antlr

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

## How to run the program

Generate java files for `ANTLR`, take note that already are present the files JavaParser on the folder main/antlr, It necessary copy the file JavaLexer from https://github.com/SoftDevSix/Argos/pull/19/files
```
./gradlew generateLexerSource generateParserSource build
```
Finally you can run tests:
```
./gradlew test
```

### CODE REVIEW

File: ASTProcessor/src/main/antlr/JavaLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,213 @@
+/*
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/main/antlr/JavaLexer.g4]: Delete the JavaLexer.g4 file

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderByText.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.antlr.v4.runtime.tree.ParseTree;
+import edu.usb.argos.ASTProcessor.antlr.JavaLexer;
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+import java.io.IOException; 
+public class FileReaderByText implements IFileAnalyzer<String, ParseTree> {
+    public ParseTree readFile(String content) throws Exception {
+        try {
+            return parseContent(content);
+        } catch (Exception e) {
+            throw new Exception("error to read file");
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderByText.java]: Use here to log errors

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

File: ASTProcessor/src/main/antlr/JavaLexer.g4 (hunk 1 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/antlr/JavaLexer.g4]: Done

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderByText.java (hunk 2 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderByText.java]: Done

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderValidation.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package main.java.edu.usb.argos.ASTProcessor.FileReaderByText;
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderValidation.java]: Move your validations class to a validations folder, the same with your other classes, divide it into folders to keep it more organized.

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderException.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package main.java.edu.usb.argos.ASTProcessor.FileReaderByText;
+
+public class FileReaderException extends Exception {
```
Comment (zamoranoalicia) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderException.java]: this should be a Runtime exception

**Review by zamoranoalicia — COMMENTED** [review]
LGTM, just a minor change request

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderValidation.java (hunk 3 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderValidation.java]: Done

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderException.java (hunk 4 above)
Comment (diego-centella-jalafund) [reply ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReaderByText/FileReaderException.java]: Done

**Review by axel-ayala-jalafund — APPROVED** [review]
Well done!

File: ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/reader/infraestructure/validation/FileReaderValidation.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package edu.usb.argos.ASTProcessor.reader.infraestructure.validation;
+
+import edu.usb.argos.ASTProcessor.reader.domain.interfaces.IFileReaderTextValidation;
+import edu.usb.argos.ASTProcessor.reader.domain.exceptions.FileReaderException;
+
+
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/reader/infraestructure/validation/FileReaderValidation.java]: remove this line space

File: ASTProcessor/src/test/java/edu/usb/argos/ASTProcessor/FileReaderTest/AntlrFileReaderByTextTest.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
+package test.java.edu.usb.argos.ASTProcessor.FileReaderTest;
+
+import edu.usb.argos.ASTProcessor.reader.application.services.FileReaderByText;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (salet-gutierrez-jalafund) [inline ASTProcessor/src/test/java/edu/usb/argos/ASTProcessor/FileReaderTest/AntlrFileReaderByTextTest.java]: It's not good to use * when importing due to we access to the all library so please import just what you need.

**Review by salet-gutierrez-jalafund — CHANGES_REQUESTED** [review]

**Pull request closed without merging**